src/routes/safe/components/Balances/index.jsx

```diff
@@ -37,196 +37,191 @@ type State = {
 }
 
 type Props = {
-  activateTokensByBalance: Function,
-  activateAssetsByBalance: Function,
-  activeTokens: List<Token>,
-  blacklistedTokens: List<Token>,
   classes: Object,
-  createTransaction: Function,
-  currencySelected: string,
-  currencyValues: BalanceCurrencyType[],
-  ethBalance: string,
-  featuresEnabled: string[],
-  fetchCurrencyValues: Function,
-  fetchTokens: Function,
-  granted: boolean,
-  safeAddress: string,
-  safeName: string,
-  tokens: List<Token>,
 }
 
 type Action = 'Token' | 'Send' | 'Receive' | 'ManageCollectibleModal'
 
-class Balances extends React.Component<Props, State> {
-  constructor(props) {
-    super(props)
-    this.state = {
-      erc721Enabled: false,
-      subMenuOptions: [],
-      showToken: false,
-      showManageCollectibleModal: false,
-      sendFunds: {
-        isOpen: false,
-        selectedToken: undefined,
-      },
-      showCoins: true,
-      showCollectibles: false,
-      showReceive: false,
-    }
-    props.fetchTokens()
-  }
+const INITIAL_STATUS: State = {
+  erc721Enabled: false,
+  subMenuOptions: [],
+  showToken: false,
+  showManageCollectibleModal: false,
+  sendFunds: {
+    isOpen: false,
+    selectedToken: undefined,
+  },
+  showCoins: true,
+  showCollectibles: false,
+  showReceive: false,
+}
 
-  static isCoinsLocation = /\/balances\/?$/
-  static isCollectiblesLocation = /\/balances\/collectibles$/
+const Balances = (props: Props) => {
+  const [state, setState] = useState(INITIAL_STATUS)
 
-  componentDidMount(): void {
-    const { activateAssetsByBalance, activateTokensByBalance, fetchCurrencyValues, safeAddress } = this.props
-    fetchCurrencyValues(safeAddress)
-    activateTokensByBalance(safeAddress)
-    activateAssetsByBalance(safeAddress)
+  const address = useSelector(safeParamAddressFromStateSelector)
+  const featuresEnabled = useSelector(safeFeaturesEnabledSelector)
 
-    const showCollectibles = Balances.isCollectiblesLocation.test(history.location.pathname)
-    const showCoins = Balances.isCoinsLocation.test(history.location.pathname)
+  useEffect(() => {
+    const isCoinsLocation = /\/balances\/?$/
```

Comment:
can this be named like `COINS_LOCATION_REGEX` and also same way for collectibles locaiton?